src/test/regress/sql/pg13.sql

```diff
@@ -115,5 +115,18 @@ EXPLAIN (ANALYZE TRUE, WAL TRUE, COSTS FALSE, SUMMARY FALSE, BUFFERS FALSE, TIMI
 WITH cte_1 AS (INSERT INTO test_wal VALUES(6,66),(7,77),(8,88) RETURNING *)
 SELECT * FROM cte_1;
 
+CREATE TABLE distributed_table(a int, b int);
+SELECT create_distributed_table('distributed_table', 'a');
+
+CREATE TABLE reference_table(a int, c int, b int);
+SELECT create_reference_table('reference_table');
+
+SELECT distributed_table.* from distributed_table JOIN reference_table ON (true);
+
+ALTER TABLE reference_table DROP COLUMN c;
+
+-- #4129
```

Comment:
I think instead of mentioning issue number, it might be better to tell what we test here

Reply:
\+ minor: still not sure if it adds any value to mention issue numbers in tests

Reply:
I feel like it is good to be able to go to the relevant issue to get more information if needed